src/lib/ngsi/ContextAttribute.cpp

```diff
@@ -57,20 +57,64 @@ using namespace orion;
 * ContextAttribute::bsonAppendAttrValue -
 *
 */
-void ContextAttribute::bsonAppendAttrValue(BSONObjBuilder& bsonAttr) const
+void ContextAttribute::bsonAppendAttrValue(BSONObjBuilder& bsonAttr, const std::string& attrType, bool autocast) const
 {
-  switch (valueType)
+  std::string effectiveStringValue = stringValue;
+  bool        effectiveBoolValue   = boolValue;
+  double      effectiveNumberValue = numberValue;
+  ValueType   effectiveValueType   = valueType;
+
+  // Checking for ValueTypeString is an additional safety measure (ensuring that the attribute came from NGSIv1 in plain text)
```

Comment:
However, if you don't look explicitly for the api version ... will this not be accepted for api v2 also?
Or, is `effectiveValueType` not set for api v2 requests?

Reply:
It is a functional requeriment NGSIv2 API be untouched. From parent issue #3112 

> NGSIv2 functionality remaing unaltered.

So, in the case you mention, a `"23"` (string) will end in the DB.

NTC (I guess)

Reply:
Eh ... sure, api v2 should NOT be touched by autocast. That's my point.

Will it be untouched ?
I say this as I see no "if (apiVersion == API_V1) ..."

Reply:
Ok, I get your point...

Note that bsonAppendAttrValue() doesn't have access to API version. However, not a problem as the caller has (in fact, the caller of the caller :) and set the `autocast` parameter properly (taking into account also the `ngsiv1Autocast` flag):

```
ngsiv1Autocast && (apiVersion == V1)
```

as can be see in this PR above.

That's helps to keep bsonAppendAttrValue() simpler.

Reply:
ok, great.

NTC